### PR TITLE
Http Request Error Mailer

### PR DESCRIPTION
**Pull request by mboneil10**
Mailer for a service change request error.

### CODE REVIEW

File: app/mailers/notifier_mailer.rb [hunk 1]
```diff
@@ -0,0 +1,9 @@
+class NotifierMailer < ApplicationMailer
+  def service_change_error(response_code)
+    CONFIG.fetch(:notification_email)
```
Comment (dfaulken) [inline app/mailers/notifier_mailer.rb]: Looks like you just called this but didn't use its value

File: app/mailers/notifier_mailer.rb [hunk 2]
```diff
@@ -0,0 +1,9 @@
+class NotifierMailer < ApplicationMailer
```
Comment (dfaulken) [inline app/mailers/notifier_mailer.rb]: Seems like we could just write

``` ruby
class NotifierMailer < ActionMailer::Base
```

File: config/environment.rb [hunk 3]
```diff
@@ -3,3 +3,5 @@
 
 # Initialize the Rails application.
 Rails.application.initialize!
+CONFIG = YAML.load_file('config/application.yml')
```
Comment (dfaulken) [inline config/environment.rb]: Could also do

``` ruby
YAML.load_file(whatever).symbolize_keys
```

File: coverage/.last_run.json [hunk 4]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 87.5
+    "covered_percent": 85.11
```
Comment (dfaulken) [inline coverage/.last_run.json]: Why's this?

File: lib/microservices_router.rb [hunk 5]
```diff
@@ -9,7 +9,11 @@ def notify_services_of_changes
                                       Service.all.to_json(include: :models))
       if request.nil?
         Net::HTTP.post_form(URI(s.url),
-                            'error' => 'an error occurred with the request')
+                            'error' => 'Request is nil')
+      elsif request == Net::HTTPSuccess
+      else
+      service_change_error(status)
```
Comment (dfaulken) [inline lib/microservices_router.rb]: Not sure if GitHub's showing the indentation weirdly, but these two lines should be indented inside the `else` block.

File: spec/mailers/previews/notifier_preview.rb [hunk 6]
```diff
@@ -0,0 +1,3 @@
+# Preview all emails at http://localhost:3000/rails/mailers/notifier
+class NotifierPreview < ActionMailer::Preview
```
Comment (dfaulken) [inline spec/mailers/previews/notifier_preview.rb]: Is this needed?

Comment (mboneil10) [reply spec/mailers/previews/notifier_preview.rb]: No

File: coverage/.last_run.json (hunk 4 above)
Comment (mboneil10) [reply coverage/.last_run.json]: Added files that aren't tested?

File: lib/microservices_router.rb (hunk 5 above)
Comment (mboneil10) [reply lib/microservices_router.rb]: Oops. I made this weird indentation.

File: app/mailers/notifier_mailer.rb [hunk 7]
```diff
@@ -0,0 +1,9 @@
+class NotifierMailer < ActionMailer::Base
+  def service_change_error(response_code)
+    notification_email = CONFIG.fetch(:notification_email)
+    mail to: notification_email,
```
Comment (dfaulken) [inline app/mailers/notifier_mailer.rb]: Sure, may as well just write it inline without setting a local variable.

File: config/application.yml [hunk 8]
```diff
@@ -0,0 +1 @@
+notification_email: 'transit-it@admin.umass.edu'
```
Comment (dfaulken) [inline config/application.yml]: Are the quotes needed in YML?

File: lib/microservices_router.rb [hunk 9]
```diff
@@ -9,7 +9,11 @@ def notify_services_of_changes
                                       Service.all.to_json(include: :models))
       if request.nil?
         Net::HTTP.post_form(URI(s.url),
-                            'error' => 'an error occurred with the request')
+                            'error' => 'Request is nil')
+      elsif request == Net::HTTPSuccess
+      else
+        service_change_error(status)
```
Comment (dfaulken) [inline lib/microservices_router.rb]: It's not clear in context from the method name that this is sending a notification.

**Review by dfaulken — CHANGES_REQUESTED** [review]

File: config/application.yml (hunk 8 above)
Comment (ennukee) [reply config/application.yml]: They're used in round-three and jobapps, so that would lead me to assume so.

Comment (dfaulken) [reply config/application.yml]: OK, but why assume? 😄 

From [wiki](https://en.wikipedia.org/wiki/YAML):

> Strings (scalars) are ordinarily unquoted, but may be enclosed in double-quotes ("), or single-quotes (').

Let's remove the quotes.

Comment (ennukee) [reply config/application.yml]: :thinking: refactor round-three and jobapps YML or riot :thinking:

Comment (ennukee) [reply config/application.yml]: on that note, `.travis.yml` files don't use quotes for string elements.

File: app/mailers/notifier_mailer.rb (hunk 7 above)
Comment (mboneil10) [reply app/mailers/notifier_mailer.rb]: Oh, right.

File: config/application.yml (hunk 8 above)
Comment (mboneil10) [reply config/application.yml]: Not sure why I added the quotes to begin with...

File: coverage/.last_run.json [hunk 10]
```diff
@@ -1,5 +1,5 @@
 {
   "result": {
-    "covered_percent": 87.5
+    "covered_percent": 84.44
```
Comment (mboneil10) [inline coverage/.last_run.json:3]: I can't figure this out. Please help @dfaulken

Comment (mboneil10) [reply coverage/.last_run.json:3]: My guess is that the files I recently deleted (mailer preview and application mailer) had a part in the coverage.

**Review by dfaulken — APPROVED** [review]

**Pull request merged**